### PR TITLE
Added method to randomise speed by +/- 30% to prevent pets from overlapping so much

### LINKED ISSUE #183: Add a random modifier (+/- 30%) to pet speed
In order to randomise the speed of pets, the speed (0-5) should be randomised by +/- 30%.

For example, when the pet speed is PetSpeed.normal (3) then a pet speed could be 

3.13, 3.04, 2.95, 3.25

Speed of 0 should still be 0, because this is the rock, which cannot move.

https://github.com/tonybaloney/vscode-pets/blob/fb11781940662b58f862b7b613794d14ce3e0b39/src/panel/pets.ts#L224

### PR DESCRIPTION
**Pull request by jyorien**
Fixes #183 and #180

### CODE REVIEW

File: src/panel/pets.ts [hunk 1]
```diff
@@ -221,7 +221,7 @@ abstract class BasePetType implements IPetType {
         this.currentState = resolveState(this.currentStateEnum, this);
 
         this._name = name;
-        this._speed = speed; // TODO #183 : Add a random modifier (+/- 30%) to this value.
+        this._speed = this.randomiseSpeed(speed);
```
Comment (tonybaloney) [inline src/panel/pets.ts:224]: please change to the American English spelling of randomize.

Even though I'm British, this repository uses American English spelling!

**Review by tonybaloney — CHANGES_REQUESTED** [review]

File: src/panel/pets.ts [hunk 2]
```diff
@@ -285,6 +285,13 @@ abstract class BasePetType implements IPetType {
         return this._speed;
     }
 
+    randomiseSpeed(speed: number) : number {
+        let min = speed * 0.7;
```
Comment (tonybaloney) [inline src/panel/pets.ts:289]: Use const instead of let

File: src/panel/pets.ts [hunk 3]
```diff
@@ -285,6 +285,13 @@ abstract class BasePetType implements IPetType {
...
+        let max = speed * 1.3;
```
Comment (tonybaloney) [inline src/panel/pets.ts:290]: And here

File: src/panel/pets.ts [hunk 4]
```diff
@@ -285,6 +285,13 @@ abstract class BasePetType implements IPetType {
...
+        let newSpeed = Math.random() * (max - min) + min;
```
Comment (tonybaloney) [inline src/panel/pets.ts:291]: And here

File: src/panel/pets.ts [hunk 5]
```diff
@@ -285,6 +285,13 @@ abstract class BasePetType implements IPetType {
...
+        return newSpeed
```
Comment (tonybaloney) [inline src/panel/pets.ts:292]: Add a semicolon to the end of this line

**Comment by jyorien** [comment]
thanks for reviewing and I have made the necessary changes!

**Review by tonybaloney — APPROVED** [review]

**Comment by tonybaloney** [comment]
Looks great! good luck with the rest of hacktoberfest

**Comment by tonybaloney** [comment]
screenshot 2022-10-07 at 17 05 57
Tested with lots of cats, works really nicely

**Pull request merged**